sabnzbd/assembler.py

```diff
@@ -169,6 +170,10 @@ def assemble(nzo: NzbObject, nzf: NzbFile, file_done: bool):
         1) Partial write: write what we have
         2) Nothing written before: write all
         """
+        # Run at most once per second unless the file is finished.
+        if nzf.next_assemble_time > time.time() and not file_done:
+            return
+        nzf.next_assemble_time = time.time() + 1
```

Comment:
Shouldn't this be much better here? So it never even enters the queue?
https://github.com/sabnzbd/sabnzbd/blob/5e58fdf82141fcec6b7b379c957909d82aae3e51/sabnzbd/nzbqueue.py#L758-L765

Reply:
I didn't realize that things were put in the assembler queue so selectively. Now I see why 5 may be enough. However, doesn't that also mean that the timing between when it's put in the queue and when it's saved can be pretty skewed? Maybe it doesn't really matter.